internal/internal_task_handlers.go

```diff
@@ -681,11 +681,11 @@ func (wth *workflowTaskHandlerImpl) getOrCreateWorkflowContext(
 		workflowContext = getWorkflowContext(runID)
 	}
 
+	executionRuntimeType := workflowCategorizedByTimeout(workflowContext)
+	metricsScope = metricsScope.Tagged(map[string]string{tagWorkflowRuntimeLength: executionRuntimeType})
```

Comment:
this will affect the deferred gauge, among other things - doesn't seem like something we want to change here.

(which looks also true for line 688, so let's change that here too / ASAP)

Reply:
good catch on the defer. Just changed the deferred func to accept tally scope as argument so it will work without this new tag as before